.travis.yml

```diff
@@ -35,11 +35,34 @@ install:
   - nvm use 8.9.0
   - npm install -g npm@'5.6.0'
 
-script:
-  - while sleep 9m; do echo "=====[ $SECONDS seconds still running ]====="; done &
-  - ./gradlew build --console=plain --stacktrace -scan -x createJavadoc -Dorg.gradle.parallel=false
-  # Killing background sleep loop
-  - kill %1
+jobs:
+  include:
+    - stage: "Build and check"
+      name: "Build without checks and tests"
+      script:
+      - while sleep 9m; do echo "=====[ $SECONDS seconds still running ]====="; done &
+      - ./gradlew build --console=plain -x check -x test --stacktrace -scan
+      # Killing background sleep loop
+      - kill %1
+    - script:
+        - while sleep 9m; do echo "=====[ $SECONDS seconds still running ]====="; done &
+        - ./gradlew --console=plain check -x test -x createJavadoc --stacktrace -scan
+        # Killing background sleep loop
+        - kill %1
+      name: "Run checks"
+    - stage: "Tests"
+      name: "Run Build + checks + tests(without integration)"
```

Comment:
I think we should add depends on the check to "Build and check" state.

Reply:
with stages, thisis already enforced.